### ISSUE TITLE
Unknown error while authenticating (npm adduser)

### ISSUE THREAD

**Issue by MMajko** [npm/npm#8132]
The log is self-explaining. Did nothing unusual.
@ Windows 8.1, versions are specified in the log.

```
`C:\xampp\htdocs\fittable>npm adduser
Username: Majko
npm WARN Username must be lowercase
Username: majko
Password:
Email: (this IS public) hlavac95@gmail.com
npm ERR! Error: Unknown error while authenticating : -/user/org.couchdb.user:maj
ko
npm ERR!     at RegClient.<anonymous> (C:\Program Files\nodejs\node_modules\npm\
node_modules\npm-registry-client\lib\request.js:308:14)
npm ERR!     at Request._callback (C:\Program Files\nodejs\node_modules\npm\node
_modules\npm-registry-client\lib\request.js:246:65)
npm ERR!     at Request.self.callback (C:\Program Files\nodejs\node_modules\npm\
node_modules\request\request.js:236:22)
npm ERR!     at Request.emit (events.js:98:17)
npm ERR!     at Request.<anonymous> (C:\Program Files\nodejs\node_modules\npm\no
de_modules\request\request.js:1142:14)
npm ERR!     at Request.emit (events.js:117:20)
npm ERR!     at IncomingMessage.<anonymous> (C:\Program Files\nodejs\node_module
s\npm\node_modules\request\request.js:1096:12)
npm ERR!     at IncomingMessage.emit (events.js:117:20)
npm ERR!     at _stream_readable.js:944:16
npm ERR!     at process._tickCallback (node.js:442:13)
npm ERR! If you need help, you may report this *entire* log,
npm ERR! including the npm and node versions, at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! System Windows_NT 6.2.9200
npm ERR! command "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nod
ejs\\node_modules\\npm\\bin\\npm-cli.js" "adduser"
npm ERR! cwd C:\xampp\htdocs\fittable
npm ERR! node -v v0.10.36
npm ERR! npm -v 1.4.28
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     C:\xampp\htdocs\fittable\npm-debug.log
npm ERR! not ok code 0

C:\xampp\htdocs\fittable>
```

**Comment by jaiew** [comment]
I had the same issue and I was on node 0.10.22.
I just updated to 0.12.2 and it worked.

**Comment by MMajko** [comment]
I used wrong login name - that was the problem. But the error shouldn't be "unknown" :)

**Comment by assertnotnull** [comment]
I confirm the problem. I am using my correct username, password and email but it fails. Even with last version.

```
13 verbose adduser back [ { [Error: Unknown error while authenticating : -/user/org.couchdb.user:assertnotnull] statusCode: 400, code: 'E400' },
13 verbose adduser   { ok: false, error: 'Unknown error while authenticating' },
13 verbose adduser   '{"ok":false,"error":"Unknown error while authenticating"}' ]
14 verbose stack Error: Unknown error while authenticating : -/user/org.couchdb.user:assertnotnull
```

**Comment by soichih** [comment]
I am having the same problem - using node 0.10.33 (from EPEL for EL6).

I can still login to www.npmjs.com, so I am sure I am entering the correct user/pass.

```
2015-04-29 22:57:43 UTC [hayashis@portal-dev1:~/git/passport-iucas]$ npm adduser
Username: soichih
Password:
Email:
Email: soichih@gmail.com
npm http PUT https://registry.npmjs.org/-/user/org.couchdb.user:soichih
npm http 400 https://registry.npmjs.org/-/user/org.couchdb.user:soichih
npm ERR! Error: Unknown error while authenticating : -/user/org.couchdb.user:soichih
npm ERR!     at RegClient.<anonymous> (/usr/lib/node_modules/npm-registry-client/lib/request.js:272:14)
npm ERR!     at Request.self.callback (/usr/lib/node_modules/request/index.js:148:22)
npm ERR!     at Request.emit (events.js:98:17)
npm ERR!     at Request.<anonymous> (/usr/lib/node_modules/request/index.js:891:14)
npm ERR!     at Request.emit (events.js:117:20)
npm ERR!     at IncomingMessage.<anonymous> (/usr/lib/node_modules/request/index.js:842:12)
npm ERR!     at IncomingMessage.emit (events.js:117:20)
npm ERR!     at _stream_readable.js:943:16
npm ERR!     at process._tickCallback (node.js:419:13)
npm ERR! If you need help, you may report this log at:
npm ERR!     <http://github.com/isaacs/npm/issues>
npm ERR! or email it to:
npm ERR!     <npm-@googlegroups.com>

npm ERR! System Linux 2.6.32-504.8.1.el6.x86_64
npm ERR! command "node" "/usr/bin/npm" "adduser"
npm ERR! cwd /home/hayashis/git/passport-iucas
npm ERR! node -v v0.10.33
npm ERR! npm -v 1.3.6
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /home/hayashis/git/passport-iucas/npm-debug.log
npm ERR! not ok code 0
```

**Comment by MMajko** [comment]
Oh. Maybe the authentication server is down, I guess? Because when I entered correct credentials, few minutes after the first attempt.

**Comment by soichih** [comment]
I just noticed this https://github.com/npm/npm/issues/7876
> Unfortunately, publishing no longer works with versions of npm older than npm@1.4.0

This is sad, because the latest version of npm published via EPEL6 is 1.3.6..

**Comment by othiym23** [comment]
> Maybe the authentication server is down, I guess?

There's no authentication service independent of the npm registry. We're still sorting out some issues related to the transition to the new registry architecture (like improving the cryptic error messages), but in general we haven't seen any availability issues.

> This is sad, because the latest version of npm published via EPEL6 is 1.3.6..

```
tag v1.4.0
Tagger: isaacs <i@izs.me>
Date:   Wed Feb 12 18:10:24 2014 -0800

v1.4.0
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.4.13 (Darwin)

iQEcBAABAgAGBQJS/CmQAAoJELCniwpsSBz2dJ0IAICcDJZMCZ51AOXv6bPBSSEz
Q0UXgcAwRaow19+9TXZwZ0Cq1wKCH+LewZX+nURO73lKrcbU5PRNsQcaaoz6bD2N
vANQFBTen8G7EG7StOROPaQBC6BxycxgTMN8pIrrvsw9VbELIeaxknDgw9phao3O
CfYRL8LlcXQODMfLEaZRWd8AfHJz96/82XmQqz6V43Ja/JwxpXraevTgc1C+stx3
h5AwE+DeUwIPEXNNSIkJxz67f2Hh7GWf73XMbXNzY0MqkyhdmvSv+TTiYhk5VrZh
KSeuO8izTL/azHvlfpH7R5qEctOQXlF9kiLH87x5VofdvWFT+sz4an+GRI7BoIw=
=m1iH
-----END PGP SIGNATURE-----
```

I know that enterprise Linux distros and language package managers exist on different time scales, but it's been well over a year since `npm@1.4.0` was released. If the maintainers responsible for Node and npm in EPEL 6 can't be persuaded to upgrade to a newer version of npm, it's always possible to use npm to install a newer version of npm (`sudo npm -g install npm`), and then use that one instead. The current version of npm (`npm@2.8.4` at this point) should work fine with any version of Node from 0.8 on up.

Closing as resolved.
